src/Network/Pusher/Data.hs

```diff
@@ -59,26 +60,54 @@ type AppSecret = B.ByteString
 
 -- |All the required configuration needed to interact with the API.
 data Pusher = Pusher
-  { pusherHost :: T.Text
-  , pusherPath :: T.Text
-  , pusherCredentials :: Credentials
+  { pusherHost              :: T.Text
+  , pusherPath              :: T.Text
+  , pusherCredentials       :: Credentials
   , pusherConnectionManager :: Manager
   }
 
 -- |The credentials for the current app.
 data Credentials = Credentials
-  { credentialsAppID :: AppID
-  , credentialsAppKey :: AppKey
+  { credentialsAppID     :: AppID
+  , credentialsAppKey    :: AppKey
   , credentialsAppSecret :: AppSecret
+  , credentialsCluster   :: Maybe Cluster
   }
 
 instance A.FromJSON Credentials where
   parseJSON (A.Object v) = Credentials
     <$> v .: "app-id"
     <*> (encodeUtf8 <$> v .: "app-key")
     <*> (encodeUtf8 <$> v .: "app-secret")
+    <*> v.:? "app-cluster"
   parseJSON v2 = failExpectObj v2
 
+-- |The cluster the current app resides on.
+data Cluster
```

Comment:
Pusher are creating new clusters fairly frequently. Using a closed list like this will require too much maintenance (e.g. we now also have a public `us2` cluster). Pusher also has private clusters on request, for which this approach won't work. I'd suggest using a string, wrapped in a `newtype`, with some standard ones listed, e.g. `eu = ClusterId "eu"`.